package.json

```diff
@@ -62,13 +63,30 @@
     "eslint-plugin-no-unsanitized": "4.0.2",
     "eslint-plugin-security": "1.5.0",
     "eslint-plugin-xss": "0.1.12",
-    "nodemon": "2.0.22"
+    "jest": "^29.7.0",
+    "nodemon": "2.0.22",
+    "ts-jest": "^29.2.5"
   },
   "overrides": {
     "semver": "7.5.3",
     "optionator": "0.9.3",
     "qs": "6.11.2",
     "systeminformation": "5.21.7",
     "axios": "1.6.1"
+  },
+  "jest": {
+    "preset": "ts-jest",
+    "testEnvironment": "node",
+    "testMatch": [
+      "**/test/unit/**/*.test.ts"
+    ],
+    "moduleFileExtensions": [
+      "ts",
+      "js",
+      "json"
+    ],
+    "collectCoverageFrom": [
+      "src/**/*.{ts,js}"
+    ]
   }
```

Comment:
We keep this in a separate `jest.config.js` file in our other repos see shardeum https://github.com/shardeum/shardeum/blob/dev/jest.config.js